src/main/java/sirius/pasta/noodle/compiler/TypeTools.java

```diff
@@ -226,26 +223,21 @@ private void reduceTable() {
      * @return the simplified type
      */
     public Type simplify(Type type) {
-        if (type instanceof TypeVariable<?>) {
-            TypeVariable<?> typeVariable = (TypeVariable<?>) type;
+        if (type instanceof TypeVariable<?> typeVariable) {
             return typeTable.getOrDefault(getTypeVariableKey(getScopeOfType(typeVariable), typeVariable.getName()),
                                           type);
         }
-        if (type instanceof WildcardType) {
-            if (((WildcardType) type).getLowerBounds().length > 0) {
-                return simplify(((WildcardType) type).getLowerBounds()[0]);
-            }
+        if ((type instanceof WildcardType) && ((WildcardType) type).getLowerBounds().length > 0) {
+            return simplify(((WildcardType) type).getLowerBounds()[0]);
         }
```

Comment:
Not sure if
```
if ((type instanceof WildcardType wildcardType) && wildcardType.getLowerBounds().length > 0 )
```
works, but if it does it would be nice

Reply:
works!